Remove deprecated code for release v0.6.0

dapomeroy opened the pull request:
Update version and remove deprecated code for release v0.6.0

dapomeroy committed:
Remove deprecated code for release v0.6.0

linsword13 commented on lib/ramble/ramble/test/end_to_end/known_applications.py:
Not sure if this is something we want to remove. I also don't quite know what the deprecation annotation refers to.

dapomeroy replied:
The deprecation tag was added here: https://github.com/GoogleCloudPlatform/ramble/commit/0c5219e108850d54ef8e1e4b6d3107d0a4c25dba
I had thought that known_applications was rendered obsolete when Doug reworked testing in cloud build, but perhaps this tag is referring to `if package_manager == "None":` on L53?

douglasjacobsen replied:
@linsword13 is right, we do not want this removed either. This is one of our most important tests.

dapomeroy replied:
Fixed this, restored the test and removed the tag.

douglasjacobsen commented on lib/ramble/ramble/test/application_language.py:
```diff
@@ -19,22 +18,6 @@
 
-@deprecation.fail_if_not_removed
-@pytest.mark.parametrize("app_class", app_types)
-def test_application_type_features(app_class):
```
I don't think this should be removed. I think we just remove the deprecation tag on this, if the `SpackApplication` is still here.

dapomeroy replied:
Updated to just remove the tag

douglasjacobsen commented on lib/ramble/ramble/test/modifier_language.py:
```diff
@@ -17,28 +16,6 @@
 mod_types = [ModifierBase, BasicModifier]  # noqa: F405
 
 
-@deprecation.fail_if_not_removed
```
Same comment here about not removing this test.

dapomeroy replied:
Updated to just remove the tag

douglasjacobsen reviewed (changes requested):
Also, if you can merge the newest develop in, that would be great.

dapomeroy commented:
This is updated, tests are passing, and I've rebased on develop. Please review the deletions again and let me know if you spot any other issues

dapomeroy committed:
Update Spack dict references to software dict

Remove support for previously deprecated Spack dictionary and update
references to software dictionary. Changes from warning to error
when a Spack dictionary is used.
